src/Core/Repositories/ExpandedPackageRepository.cs

```diff
@@ -95,7 +98,12 @@ public IEnumerable<IPackage> FindPackagesById(string packageId)
                 if (SemanticVersion.TryParse(versionDirectoryName, out version) &&
                     Exists(packageId, version))
                 {
-                    yield return GetPackageInternal(packageId, version);
+                    var package = GetPackageInternal(packageId, version);
+                    if (package == null)
+                    {
+                        continue;
+                    }
+                    yield return package;
                 }
             }
```

Comment:
I wonder why this hasn't had a yield break;  I hope no callers are relying on this to return null in some odd situations.